cmd/state-svc/internal/resolver/analytics/analytics.go

```diff
@@ -0,0 +1,234 @@
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/graph"
+	"github.com/ActiveState/cli/internal/installation/storage"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/machineid"
+	"github.com/ActiveState/cli/internal/singleton/uniqid"
+	"github.com/ActiveState/cli/internal/updater"
+	ga "github.com/ActiveState/go-ogle-analytics"
+	"github.com/ActiveState/sysinfo"
+)
+
+// AnalyticsResolver resolves requests to forward analytics events
+type Resolver struct {
+	gaClient         *ga.Client
+	customDimensions customDimensions
+	closer           func()
+	ctx              context.Context
+	events           chan event.EventData
+}
+
+// NewResolver initializes the resolver starting an event loop from which events are sent to the backend and initializing all the static custom dimensions
+func NewResolver(cfg *config.Instance) *Resolver {
+	installSource, err := storage.InstallSource()
+	if err != nil {
+		logging.Error("Could not detect installSource: %s", errs.Join(err, " :: ").Error())
+	}
+
+	id := machineid.UniqID()
+	var trackingID string
+	if !condition.InUnitTest() {
+		trackingID = constants.AnalyticsTrackingID
+	}
+
+	client, err := ga.NewClient(trackingID)
+	if err != nil {
+		logging.Error("Cannot initialize analytics: %s", err.Error())
+		client = nil
+	}
+
+	if client != nil {
+		client.ClientID(id)
+	}
+
+	osName := sysinfo.OS().String()
+	osVersion := "unknown"
+	osvInfo, err := sysinfo.OSVersion()
+	if err != nil {
+		logging.Errorf("Could not detect osVersion: %v", err)
+	}
+	if osvInfo != nil {
+		osVersion = osvInfo.Version
+	}
+
+	sessionToken := cfg.GetString(anaConsts.CfgSessionToken)
+	tag, ok := os.LookupEnv(constants.UpdateTagEnvVarName)
+	if !ok {
+		tag = cfg.GetString(updater.CfgUpdateTag)
+	}
+
+	customDimensions := customDimensions{
+		version:       constants.Version,
+		branchName:    constants.BranchName,
+		osName:        osName,
+		osVersion:     osVersion,
+		installSource: installSource,
+		machineID:     machineid.UniqID(),
+		uniqID:        uniqid.Text(),
+		sessionToken:  sessionToken,
+		updateTag:     tag,
+	}
+
+	if id == "unknown" {
+		logging.Error("unknown machine id")
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	r := &Resolver{
+		gaClient:         client,
+		customDimensions: customDimensions,
+		ctx:              ctx,
+		events:           make(chan event.EventData),
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer handlePanics(recover(), debug.Stack())
+		defer wg.Done()
+		r.eventLoop()
+	}()
+
+	r.closer = func() {
+		cancel()
+		wg.Wait()
+	}
+
+	return r
+}
+
+// AnalyticsEvent schedules the sending of the analytics event
+func (r *Resolver) AnalyticsEvent(_ context.Context, category, action string, label, projectName, out, userID *string) (*graph.AnalyticsEventResponse, error) {
```

Comment:
Let's move this into the resolver package. 

I think most of the implementation detail you have in this package has nothing to do with "resolving". The resolving layer should be lightweight and simple. It should direct the request to the lower level implementation, which would be this package (though it probably shouldn't live under the `resolver` package at all).